src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockFetcher.scala

```diff
@@ -301,37 +299,38 @@ class BlockFetcher(
     log.debug("Fetching bodies")
 
     val hashes = state.takeHashes(syncConfig.blockBodiesPerRequest)
-    requestBlockBodies(hashes) pipeTo self
+    requestBlockBodies(hashes).runToFuture pipeTo self
   }
 
   private def fetchStateNode(hash: ByteString, originalSender: ActorRef, state: BlockFetcherState): Unit = {
     log.debug("Fetching state node for hash {}", ByteStringUtils.hash2string(hash))
-    requestStateNode(hash, originalSender) pipeTo self
+    requestStateNode(hash).runToFuture pipeTo self
     val newState = state.fetchingStateNode(hash, originalSender)
 
     context become started(newState)
   }
 
-  private def requestBlockHeaders(msg: GetBlockHeaders): Future[Any] =
+  private def requestBlockHeaders(msg: GetBlockHeaders): Task[Any] =
     makeRequest(Request.create(msg, BestPeer), RetryHeadersRequest)
       .flatMap {
         case Response(_, BlockHeaders(headers)) if headers.isEmpty =>
           log.debug("Empty BlockHeaders response. Retry in {}", syncConfig.syncRetryInterval)
-          Future.successful(RetryHeadersRequest).delayedBy(syncConfig.syncRetryInterval)
-        case res => Future.successful(res)
+          Task.now(RetryHeadersRequest).delayResult(syncConfig.syncRetryInterval)
+        case res => Task.now(res)
       }
 
-  private def requestBlockBodies(hashes: Seq[ByteString]): Future[Any] =
+  private def requestBlockBodies(hashes: Seq[ByteString]): Task[Any] =
     makeRequest(Request.create(GetBlockBodies(hashes), BestPeer), RetryBodiesRequest)
 
-  private def requestStateNode(hash: ByteString, requestor: ActorRef): Future[Any] =
+  private def requestStateNode(hash: ByteString): Task[Any] =
```

Comment:
I would keep this ActorRef, at least for now. WDYT?

Reply:
It's an unused parameter, I don't think it makes sense to keep it

Reply:
Make sense 👍